survey_five_stars_question_type/models/survey_user_input_line.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SurveyUser_input_line(models.Model):
+
+    _inherit = 'survey.user_input_line'
+
```

Comment:
Shouldn't we add a python constrains which check that the answer is between 1 and 5?

Reply:
Already done at `models/survey_question.py: validate_star_rate`

Reply:
I'm talking about a python model constrain, because if I do a simple ```write``` with an invalid value, there will be no error.

Reply:
Yes I agree, a simple `write` bypass the constraint, we may have a question type matrix and as answer we put text.
But it's done like that in survey module, the check is done only at user submit.
A method `save_{question_type}` call `validate_{question_type}`.

Reply:
Odoo doesn't use pylint/flake8 but we do :smile:
We might need more opinion about that but I would like to have a constrain for that. I would say, it isn't because Odoo did it wrong that we should follow the same path ^^.